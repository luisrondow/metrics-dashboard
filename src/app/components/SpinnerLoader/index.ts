export { default } from './spinner-loader.component';