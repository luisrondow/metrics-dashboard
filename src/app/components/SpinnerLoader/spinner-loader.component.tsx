import { Spinner } from './spinner-loader.styles';

const SpinnerLoader = () => <Spinner />;

export default SpinnerLoader;
