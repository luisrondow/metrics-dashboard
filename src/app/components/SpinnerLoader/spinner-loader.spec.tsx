import { render, screen } from '@testing-library/react';

import SpinnerLoader from './spinner-loader.component';

jest.mock('./spinner-loader.styles', () => {
  const Spinner = () => <div data-testid="spinner"></div>;

  return { Spinner };
});

describe('SpinnerLoader Component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<SpinnerLoader />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders spinner correctly', () => {
    render(<SpinnerLoader />);
    const spinnerElement = screen.getByTestId('spinner');
    expect(spinnerElement).toBeInTheDocument();
  });
});
