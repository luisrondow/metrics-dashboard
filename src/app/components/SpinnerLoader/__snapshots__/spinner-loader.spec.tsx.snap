// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpinnerLoader Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="spinner"
  />
</DocumentFragment>
`;
